src/carbonio-files-ui-common/hooks/useAccessCode.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { useCallback, useMemo, useState } from 'react';
+
+import { generateAccessCode } from '../utils/utils';
+
+export function useAccessCode(
+	isAccessCodeEnabledInitialState: boolean,
+	newAccessCodeValueInitialState?: string | null
+): {
+	newAccessCodeValue: string;
+	isAccessCodeEnabled: boolean;
+	toggleAccessCode: () => void;
+	regenerateAccessCode: () => void;
+	reset: () => void;
+	resetToInitial: () => void;
```

Comment:
Are two reset really needed? At the moment reset is only used in the add section, where the hook is initialized with false and undefined. Aka reset behavior is the same of resetToInitial with the initial values false and undefined.